pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -199,32 +214,195 @@ pub mod pallet {
+				// we assume all currencies that already had their total issuance at 0 have already been deactivated in a previous step
+				for (currency_id, _) in total_issuances {
+					T::Fungibles::start_destroy(currency_id, None)?;
+				}
+				Self::deposit_event(Event::RefundCompleted(pool_id.clone()));
+			} else {
+				// deactivate all currencies whose issuance dropped to 0 in this step
+				for currency_id in dead_currencies {
+					T::Fungibles::start_destroy(currency_id.clone(), None)?;
+				}
+			}
+
+			Ok(())
 		}
 
 		#[pallet::call_index(9)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn finish_destroy(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn finish_destroy(origin: OriginFor<T>, pool_id: T::PoolId) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let pool_details = Pools::<T>::get(pool_id.clone()).ok_or(Error::<T>::PoolUnknown)?;
+
+			ensure!(pool_details.state.is_destroying(), Error::<T>::InUse);
+
+			for currency_id in pool_details.bonded_currencies {
+				if T::Fungibles::asset_exists(currency_id.clone()) {
+					// This would fail with an InUse error if there are any accounts left on any currency
+					T::Fungibles::finish_destroy(currency_id.clone())?;
+				}
+			}
+
+			let pool_account = pool_id.clone().into();
+
+			let total_collateral_issuance = Self::get_pool_collateral(&pool_account);
+
+			if !total_collateral_issuance.is_zero() {
+				T::CollateralCurrency::transfer(
+					T::CollateralAssetId::get(),
+					&pool_account,
+					&pool_details.manager,
+					total_collateral_issuance,
+					Preservation::Expendable,
+				)?;
+			}
+
+			Pools::<T>::remove(&pool_id);
+
+			// TODO: refund deposit
+
+			Self::deposit_event(Event::Destroyed(pool_id));
+
+			Ok(())
 		}
 	}
 
-	impl<T: Config> Pallet<T> {}
+	impl<T: Config> Pallet<T> {
+		fn do_start_destroy_pool(pool_id: T::PoolId, maybe_check_owner: Option<AccountIdOf<T>>) -> DispatchResult {
+			let mut pool_details = Pools::<T>::get(pool_id.clone()).ok_or(Error::<T>::PoolUnknown)?;
+
+			ensure!(!pool_details.state.is_destroying(), Error::<T>::Destroying);
+
+			if let Some(caller) = maybe_check_owner {
+				ensure!(pool_details.is_manager(&caller), Error::<T>::Unauthorized);
+			}
+
+			pool_details.state.destroy();
+
+			Self::deposit_event(Event::DestructionStarted(pool_id.clone()));
+
+			let total_collateral_issuance = Self::get_pool_collateral(&pool_id.clone().into());
+
+			if total_collateral_issuance.is_zero() {
+				// deactivate all currencies and emit RefundCompleted
+				for currency_id in pool_details.bonded_currencies.iter() {
+					T::Fungibles::start_destroy(currency_id.clone(), None)?;
+				}
+
+				Self::deposit_event(Event::RefundCompleted(pool_id.clone()));
+
+				return Ok(());
+			}
+
+			let mut destroyed: usize = 0; 
+			for currency_id in pool_details.bonded_currencies.iter() {
+				if T::Fungibles::total_issuance(currency_id.clone()).is_zero() {
```

Comment:
Else?

Reply:
If a currency's issuance is non-zero, we can't move it to destroying state, or else we won't be able to perform burns on it anymore.

Reply:
So no else